File: Python3/프로그래머스/0/120956. 옹알이 （1）/옹알이 （1）.py
def solution(babbling):
    answer = 0
    for i in babbling:
        k = i.replace("aya", " ").replace("ye", " ").replace("ma", " ").replace("woo", " ").strip()
        if k == "":
            answer = answer + 1
            
    return answer

File: C++/프로그래머스/0/120956. 옹알이 （1）/옹알이 （1）.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

static const vector<string> samples = {"aya", "ye", "woo", "ma"};

bool can_speak(const string& s) {
    string t = s;
    for (const string& w : samples) {
        size_t p = t.find(w);
        if (p != string::npos) t.replace(p, w.size(), " ");
    }
    t.erase(remove(t.begin(), t.end(), ' '), t.end());
    return t.empty();
}

int solution(vector<string> babbling) {
    int answer = 0;
    
    for (const string& str : babbling) {
        if (can_speak(str)) answer++;
    }
    
    return answer;
}